src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb

```diff
@@ -222,26 +264,75 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
                 Return valueText
             End Function
 
-            Private Shared Function CreateFloatingLiteralToken(token As SyntaxToken, newValueString As String, newValue As Double) As SyntaxToken
-                Debug.Assert(token.Kind = SyntaxKind.FloatingLiteralToken)
+            Private Function GetIntegerLiteralValueString(value As Object, base As LiteralBase) As String
+                Select Case base
+                    Case LiteralBase.Decimal
+                        Return CType(value, ULong).ToString(CultureInfo.InvariantCulture)
+                    Case LiteralBase.Hexadecimal
+                        Dim valueString = ""
+                        If TypeOf (value) Is Short Then
+                            valueString = CType(value, Short).ToString("X")
+                        ElseIf TypeOf (value) Is Integer Then
+                            valueString = CType(value, Integer).ToString("X")
+                        ElseIf TypeOf (value) Is Long Then
+                            valueString = CType(value, Long).ToString("X")
+                        Else
+                            valueString = CType(value, ULong).ToString("X")
+                        End If
 
+                        Return "&H" & valueString
+                    Case LiteralBase.Octal
+                        Dim val1 As ULong
+                        If TypeOf (value) Is Short Then
+                            val1 = CType(value, UShort)
+                        ElseIf TypeOf (value) Is Integer Then
+                            val1 = CType(value, UInteger)
+                        ElseIf TypeOf (value) Is Long Then
+                            val1 = CType(value, ULong)
+                        Else
+                            val1 = CType(value, ULong)
```

Comment:
@AnthonyDGreen, is there a more VB-ish way to do this? I'm always terrified of CType.